paradox-nakadi-consumer-boot/src/main/java/de/zalando/paradox/nakadi/consumer/boot/FailedEventReplayer.java

```diff
@@ -0,0 +1,108 @@
+package de.zalando.paradox.nakadi.consumer.boot;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Collection;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+
+import org.springframework.stereotype.Component;
+
+import de.zalando.paradox.nakadi.consumer.boot.components.EventReceiverRegistry;
+import de.zalando.paradox.nakadi.consumer.boot.components.EventTypeConsumer;
+import de.zalando.paradox.nakadi.consumer.boot.components.FailedEventSourceMap;
+import de.zalando.paradox.nakadi.consumer.core.EventHandler;
+import de.zalando.paradox.nakadi.consumer.core.FailedEventSource;
+import de.zalando.paradox.nakadi.consumer.core.domain.EventTypePartition;
+import de.zalando.paradox.nakadi.consumer.core.domain.FailedEvent;
+
+@Component
+public class FailedEventReplayer {
+
+    private final EventReceiverRegistry eventReceiverRegistry;
+
+    private final FailedEventSourceMap failedEventSourceMap;
+
+    private final ReplayHandler replayHandler;
+
+    private static final FailedEventSource<FailedEvent> NON_EXIST_FAILED_EVENT_SOURCE =
+        new FailedEventSource<FailedEvent>() {
+
+            private static final String EVENT_SOURCE_NAME_IS_NOT_AVAILABLE_MESSAGE =
+                "Event source name is not available.";
+
+            @Override
+            public Optional<FailedEvent> getFailedEvent() {
+                throw new IllegalArgumentException(EVENT_SOURCE_NAME_IS_NOT_AVAILABLE_MESSAGE);
+            }
+
+            @Override
+            public void commit(final FailedEvent failedEvent) {
+                throw new IllegalArgumentException(EVENT_SOURCE_NAME_IS_NOT_AVAILABLE_MESSAGE);
+            }
+
+            @Override
+            public String getEventSourceName() {
+                throw new IllegalArgumentException(EVENT_SOURCE_NAME_IS_NOT_AVAILABLE_MESSAGE);
+            }
+
+            @Override
+            public long getSize() {
+                throw new IllegalArgumentException(EVENT_SOURCE_NAME_IS_NOT_AVAILABLE_MESSAGE);
```

Comment:
Isn't `UnsupportedOperationException` better? Besides, `Event source name is not available.` is not a good message.